internal/gatewayapi/backendtrafficpolicy.go

```diff
@@ -345,7 +345,14 @@ func (t *Translator) translateBackendTrafficPolicyForGateway(policy *egv1a1.Back
 	// Should exist since we've validated this
 	ir := xdsIR[irKey]
 
+	policyTarget := irStringKey(
```

Comment:
can we unify the logic and style used for route matching as well ?
https://github.com/envoyproxy/gateway/blob/c30d037a0018ddb131cacb2f041ae5bc81dcae10/internal/gatewayapi/backendtrafficpolicy.go#L284

with comments on - how prefix is generated and a helper match to compute a match

Reply:
https://github.com/envoyproxy/gateway/blob/c30d037a0018ddb131cacb2f041ae5bc81dcae10/internal/gatewayapi/helpers.go#L361

Reply:
Sorry, I don't follow your meaning.